### ISSUE TITLE
fallocate06.c  failed to test in btrfs

### ISSUE THREAD

**Issue by weidongkl** [linux-test-project/ltp#804]
When I test this case by using btrfs, this case failed as below:
TFAIL: fallocate(FALLOC_FL_PUNCH_HOLE | FALLOC_FL_KEEP_SIZE) failed unexpectedly: ENOSPC (28)
env:
```
[root@localhost ltp]# uname -m
aarch64
[root@localhost ltp]# uname -r
4.19.90-2012.5.0.0054.oe1.aarch64
```
command
```
./runltp -f syscalls -s fallocate06
```
error in btrfs
```
tst_test.c:1329: TINFO: Testing on btrfs
tst_test.c:859: TINFO: Formatting /dev/loop0 with btrfs opts='' extra opts=''
tst_test.c:1263: TINFO: Timeout per run is 0h 05m 00s
fallocate06.c:117: TINFO: Copy-on-write is supported
fallocate06.c:168: TINFO: Case 1. Fill FS: no; Use copy on write: no
fallocate06.c:157: TPASS: write() successful
fallocate06.c:201: TPASS: Misaligned allocation works as expected
fallocate06.c:157: TPASS: fallocate(FALLOC_FL_PUNCH_HOLE | FALLOC_FL_KEEP_SIZE) successful
fallocate06.c:237: TPASS: fallocate(FALLOC_FL_PUNCH_HOLE | FALLOC_FL_KEEP_SIZE) cleared the correct file range
fallocate06.c:168: TINFO: Case 2. Fill FS: yes; Use copy on write: no
tst_fill_fs.c:32: TINFO: Creating file mntpoint/file0 size 21710183
tst_fill_fs.c:32: TINFO: Creating file mntpoint/file1 size 8070086
tst_fill_fs.c:32: TINFO: Creating file mntpoint/file2 size 3971177
tst_fill_fs.c:32: TINFO: Creating file mntpoint/file3 size 36915315
tst_fill_fs.c:32: TINFO: Creating file mntpoint/file4 size 70310993
tst_fill_fs.c:32: TINFO: Creating file mntpoint/file5 size 4807935
tst_fill_fs.c:32: TINFO: Creating file mntpoint/file6 size 90739786
tst_fill_fs.c:32: TINFO: Creating file mntpoint/file7 size 76896492
tst_fill_fs.c:32: TINFO: Creating file mntpoint/file8 size 72228649
tst_fill_fs.c:32: TINFO: Creating file mntpoint/file9 size 36207821
tst_fill_fs.c:32: TINFO: Creating file mntpoint/file10 size 81483962
tst_fill_fs.c:59: TINFO: write(): ENOSPC (28)
fallocate06.c:157: TPASS: write() successful
fallocate06.c:201: TPASS: Misaligned allocation works as expected
fallocate06.c:146: TFAIL: fallocate(FALLOC_FL_PUNCH_HOLE | FALLOC_FL_KEEP_SIZE) failed unexpectedly: ENOSPC (28)
fallocate06.c:168: TINFO: Case 3. Fill FS: no; Use copy on write: yes
fallocate06.c:157: TPASS: write() successful
```

**Comment by mdoucha** [comment]
This is a known kernel bug fixed in kernel v5.9 by commit [6d4572a9d71d5fc2affee0258d8582d39859188c](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=6d4572a9d71d).

**Comment by pevik** [comment]
@weidongkl FYI Martin documented missing git commit in https://github.com/linux-test-project/ltp/commit/920a4502630d86f45afa8685b5acffda49b395c1.
